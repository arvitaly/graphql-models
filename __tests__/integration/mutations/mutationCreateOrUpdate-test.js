"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const CreateDuplicateError_1 = require("./../../../CreateDuplicateError");
const utils_1 = require("./../utils");
it("mutationCreateOrUpdate: animal", () => __awaiter(this, void 0, void 0, function* () {
    const tester = new utils_1.default();
    const animal = tester.generateModel("animal");
    tester.adapter.createOne.mockImplementation(() => {
        throw new CreateDuplicateError_1.default("");
    });
    tester.adapter.findOrCreateOne.mockReturnValue(animal);
    tester.adapter.findOne.mockReturnValueOnce(animal);
    tester.adapter.updateOne.mockReturnValue(animal);
    const result = yield tester.query(`mutation M1{
            createOrUpdateAnimal(input:{
                create:{
                    name: "Hi",
                    Weight: 1.5
                }
                update: {
                    setWeight: {Weight: 5.7}
                }
            }){
                animal{
                    id
                    Weight
                }
            }}`);
    expect(tester.adapter.createOne.mock.calls).toMatchSnapshot();
    expect(tester.adapter.updateOne.mock.calls).toMatchSnapshot();
    expect(result).toMatchSnapshot();
}));
