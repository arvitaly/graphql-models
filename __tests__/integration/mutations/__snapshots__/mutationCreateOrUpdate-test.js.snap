// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutationCreateOrUpdate: animal 1`] = `
Array [
  Array [
    "animal",
    Object {
      "Weight": 1.5,
      "name": "Hi",
    },
  ],
]
`;

exports[`mutationCreateOrUpdate: animal 2`] = `
Array [
  Array [
    "animal",
    "1",
    Object {
      "Weight": 5.7,
    },
  ],
]
`;

exports[`mutationCreateOrUpdate: animal 3`] = `
Object {
  "createOrUpdateAnimal": Object {
    "animal": Object {
      "Weight": 1001.1,
      "id": "QW5pbWFsOjE=",
    },
  },
}
`;
